upcoming:[DI-24467] - Update group_by type in widgets

ankita-akamai opened the pull request:
## Description 📝
Update group_by type in widgets.

## Changes  🔄
- Update `Widgets` and `CloudPulseMetricsRequest` in types.

## Target release date 🗓️
6 May 2025

## Preview 📷

| Before  | After   |
| ------- | ------- |
| `group_by` wasn't showing up as it was undefined ![image](https://github.com/user-attachments/assets/cac31a0f-d8ad-4c83-a1c2-156a42cbcbf3) | group_by is defined and visible in paylaod ![image](https://github.com/user-attachments/assets/353c25ca-cccd-427d-a9f6-cac609843e80) |
|group_by is of string type ![image](https://github.com/user-attachments/assets/25836dbc-44a0-4e6e-ac61-2e0ce832af84) | group_by is of string[] type ![image](https://github.com/user-attachments/assets/8588d634-c385-4f65-9d43-3f6e49df3a44) |

## How to test 🧪

### Verification steps

(How to verify changes)

- Go to Metrics page, select filters.
- Go to networks tab and observe payload of `/metrics` call.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

*Check all that apply*

- [ ] Use React components instead of HTML Tags
- [ ] Proper naming conventions like cameCase for variables & Function & snake_case for constants
- [ ] Use appropriate types & avoid using "any"
- [ ] No type casting & non-null assertions
- [ ] Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] Providing/Improving test coverage
- [ ] Use sx props to pass styles instead of style prop
- [ ] Add JSDoc comments for interface properties & functions
- [ ] Use strict equality (===) instead of double equal (==)
- [ ] Use of named arguments (interfaces) if function argument list exceeds size 2
- [ ] Destructure the props
- [ ] Keep component size small & move big computing functions to separate utility
- [ ] 📱 Providing mobile support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

---

ankita-akamai commented:
Failing UT is not related to this pr, checking cypress fail

venkymano-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/dbaas-updatewidgets-verification.spec.ts:
```diff
@@ -189,7 +190,7 @@ describe('Integration tests for verifying Cloudpulse custom and preset configura
         expect(filtersStr).to.include('"operator":"eq"');
         expect(filtersStr).to.include('"value":"secondary"');
-        expect(group_by).to.equal('region');
+        expect(group_by).to.deep.equal(['entity_id']);
```
```suggestion
        expect(group_by.length).to.equal(1);
        expect(group_by[0]).to.equal('entity_id');
```

venkymano-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/dbaas-updatewidgets-verification.spec.ts:
```diff
@@ -271,7 +272,7 @@ describe('Integration tests for verifying Cloudpulse custom and preset configura
         expect(filtersStr).to.include('"operator":"eq"');
         expect(filtersStr).to.include('"value":"primary"');
-        expect(group_by).to.equal('region');
+        expect(group_by).to.deep.equal(['entity_id']);
```
we can use same here as well

venkymano-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/dbaas-updatewidgets-verification.spec.ts:
```diff
@@ -65,6 +65,7 @@ const dashboard = dashboardFactory.build({
       metric: name,
       unit,
       y_label: yLabel,
+      group_by: ['entity_id'],
```
You can add one multiple case as well...what you say??

venkymano-akamai reviewed (approved):
@ankita-akamai , approving pending some cypress comments, please see , reply back if you don't need to take it

ankita-akamai commented:
> @ankita-akamai , approving pending some cypress comments, please see , reply back if you don't need to take it

Addressed comments, we can merge this now after checks pass... @venkymano-akamai
